include/oneapi/dpl/pstl/hetero/algorithm_ranges_impl_hetero.h

```diff
@@ -489,14 +611,27 @@ __pattern_adjacent_find(__hetero_tag<_BackendTag>, _ExecutionPolicy&& __exec, _R
 }
 
 #if _ONEDPL_CPP20_RANGES_PRESENT
+template <typename _Pred, typename _Proj>
+struct __pattern_adjacent_find_ranges_pred
```

Comment:
We already defined a type with the same functionality above. And we can re-use one.

Reply:
Fixed